### PR TITLE
plugins: update python plugin to support get_pull_properties()

### PR DESCRIPTION
**Pull request by ZenHarbinger**
Also updated schema test in unit tests.
Added tests for get_pull_properties() to test.

https://bugs.launchpad.net/snapcraft/+bug/1650626

### CODE REVIEW

File: snapcraft/tests/test_plugin_python.py [hunk 1]
```diff
@@ -257,7 +269,7 @@ def test_fileset_ignores(self):
         self.assertListEqual(expected_fileset, fileset)
 
     @mock.patch.object(python.PythonPlugin, 'run')
-    def test_build_fixes_python_shebangs(self, run_mock):
+    def test_pull_fixes_python_shebangs(self, run_mock):
```
Comment (kyrofa) [inline snapcraft/tests/test_plugin_python.py]: Why this change? The shebang rewrite is happening in build, and this test tests the build.

**Comment by ZenHarbinger** [comment]
Yes, that was an unexpected rename of the test method.
Renamed and re-committed.
